### PR TITLE
Add PII JWT and credit card details in api response

### PR DESCRIPTION
**Pull request by aryangupta701**
Fixes: SCA-565

### CODE REVIEW

File: src/app/vulnerabilities/apiKeyScanner.php [hunk 1]
```diff
@@ -14,6 +14,28 @@ function (Request $request, Response $response) {
...
+        Discover Card = 6011 1111 1111 1117<br>
+        JCB Card = 3530 1113 3330 0000<br>
+        Maestro Card = 6759 6498 2643 8453<br>
+        Master Card = 5555 5555 5555 4444<br>
+        Visa Card = 4111 1111 1111 1111<br>
+        Rupay Card = 6521 6521 6521 6521<br>
+        Password = 12345678<br>
+        Religion = Hindu<br>
+        Sexual Orientation <br>
+        Racial Background <br>
```
Comment (divyanshjainATastra) [inline src/app/vulnerabilities/apiKeyScanner.php:28]: Straight
Asian

File: src/app/vulnerabilities/apiKeyScanner.php [hunk 2]
```diff
@@ -14,6 +14,28 @@ function (Request $request, Response $response) {
...
+        Passport Number = A1234567<br>
+        Driving License Number = DL-0420110149646<br>
+        Vehicle Registration Number = DL-0420110149646<br>
+        SSN = 078-05-1120<br>
+        IBAN = DE89 3704 0044 0532 0130 00<br>
+        Username = admin<br>
+        Email = admin@gmail.com<br>
```
Comment (divyanshjainATastra) [inline src/app/vulnerabilities/apiKeyScanner.php:35]: say+my+name@bb.com kardo dono me

File: src/app/vulnerabilities/apiKeyScanner.php [hunk 3]
```diff
@@ -14,6 +14,28 @@ function (Request $request, Response $response) {
...
+        Personal-Information = 1234567890<br>
+        JWT token = eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0ZXN0IiwiaWF0Ijox <br>
```
Comment (divyanshjainATastra) [inline src/app/vulnerabilities/apiKeyScanner.php:37]: invalid jwt

File: src/app/vulnerabilities/apiKeyScanner.php [hunk 4]
```diff
@@ -14,6 +14,28 @@ function (Request $request, Response $response) {
...
         JitPack Authentication Token = jp_hjhggfhgfvdbhfsdhg32<br>
+        American Express Card = 3782 822463 10005<br>
+        Diners Club Card = 3056 9309 0259 04<br>
+        Discover Card = 6011 1111 1111 1117<br>
+        JCB Card = 3530 1113 3330 0000<br>
+        Maestro Card = 6759 6498 2643 8453<br>
+        Master Card = 5555 5555 5555 4444<br>
+        Visa Card = 4111 1111 1111 1111<br>
+        Rupay Card = 6521 6521 6521 6521<br>
+        Password = 12345678<br>
```
Comment (divyanshjainATastra) [inline src/app/vulnerabilities/apiKeyScanner.php:25]: heisenberg

File: src/app/vulnerabilities/apiKeyScanner.php [hunk 5]
```diff
@@ -14,6 +14,28 @@ function (Request $request, Response $response) {
...
         Social Security Number = 078-05-1120<br>
         JitPack Authentication Token = jp_hjhggfhgfvdbhfsdhg32<br>
+        American Express Card = 3782 822463 10005<br>
+        Diners Club Card = 3056 9309 0259 04<br>
+        Discover Card = 6011 1111 1111 1117<br>
+        JCB Card = 3530 1113 3330 0000<br>
+        Maestro Card = 6759 6498 2643 8453<br>
+        Master Card = 5555 5555 5555 4444<br>
+        Visa Card = 4111 1111 1111 1111<br>
+        Rupay Card = 6521 6521 6521 6521<br>
```
Comment (divyanshjainATastra) [inline src/app/vulnerabilities/apiKeyScanner.php:24]: Make sure this card does got validate against is_luhn function

**Comment by twelcon** [comment]
Just FYI: We have removed `Religion`, `Sexual Orientation`, `Racial Background`  from PII check as suggested by ananda. So can update the pii response accordingly.
CC: @divyanshjainATastra

**Comment by divyanshjainATastra** [comment]
> Just FYI: We have removed `Religion`, `Sexual Orientation`, `Racial Background` from PII check as suggested by ananda. So can update the pii response accordingly. CC: @divyanshjainATastra

@twelcon This kind of thing should not be processed unless user explict consent is there(GDPR guidelines). Also, did you let him about the sensitivity-tag that we utilise to check PII as per it's sensitivity. If we doesn't want to check for pii as per "main PII rule" , we can easily do that by changing value to 4. Instead of removing this key value pair

**Comment by divyanshjainATastra** [comment]
@twelcon deploy this changes

**Review by twelcon — APPROVED** [review]

**Pull request merged**